nixos/tests/matrix/synapse-workers.nix

```diff
@@ -0,0 +1,45 @@
+  import ../make-test-python.nix ({ pkgs, ... }:
+  {
+    name = "matrix-synapse-workers";
+    meta = {
+    maintainers = pkgs.matrix-synapse.meta.maintainers;
+    };
+
+    nodes = {
+      homeserver = { pkgs, nodes, ... }: {
+        services.postgresql = {
+          enable = true;
+          initialScript = pkgs.writeText "synapse-init.sql" ''
+            CREATE ROLE "matrix-synapse" WITH LOGIN PASSWORD 'synapse';
+            CREATE DATABASE "matrix-synapse" WITH OWNER "matrix-synapse"
+            TEMPLATE template0
+            LC_COLLATE = "C"
+            LC_CTYPE = "C";
+          '';
+        };
+
+        services.redis.servers."".enable = true;
+
+        services.matrix-synapse = {
+          enable = true;
+          settings = {
+            enable_registration = true;
+            enable_registration_without_verification = true;
+
+            federation_sender_instances = [ "federation_sender" ];
```

Comment:
Can't we configure that from worker.instances by setting the type of worker to federation_sender?

Reply:
No, you don't use `worker_app = federation_sender` so there's no way to know which `generic_worker`s are federation_senders. That kind of thing is exactly what I was talking about coming in the later PR